### ISSUE TITLE
Better Accessibility

### ISSUE THREAD

**Issue by aenjojo** [aenjojo/aenjojo-dev#4]
Add better accessibilities to components, so user with accessibility problem will have a great experience with screen-reader or motion-reduce.

* `sr-only` element to read by screen-reader inside main focus element without context, such as icon-only button, wrapped component inside link, etc.
* style for user preferences, such as no animation, less animation, or dark mode.
* better contrast color between text and background or between higher element (node/children) and lower element (root/parent). decoration-only element is not included.

**Comment by aenjojo** [comment]
Preferences should be styled as `motion-reduce` first and use `motion-safe` to style dynamic looks for no preferred motion user.

**Comment by aenjojo** [comment]
According to Web Content Accessibility Guidelines (WCAG), an image should contain a descriptive alternative text (`alt`) if the image correlated with the content and explained the content.

Decoration-only image, includes image that correlated but not to explain the content, must use empty string `alt`.

Description contained inside `alt` text must explain the image well and must not repeated on the caption or the content.

**Comment by aenjojo** [comment]
As reported by [WAVE](https://wave.webaim.org/report#/https://aenjojo.dev/) and [W3C Validator](https://validator.w3.org/nu/?doc=https%3A%2F%2Faenjojo.dev%2F) tools for the website, here are some reports that need a workaround with its solutions:
<details>
<summary>Report 1:

Navigation should be differentiated by giving a name

</summary>

> Solution:
> * Give each `<nav>` a name property
> * Assign `desktop` for desktop view navigation
> * Assign `mobile` for mobile view navigation

</details>

<details>
<summary>Report 2:

Linked images that already have text in `<a>` should not have `alt` or `title` property in `<img>`

</summary>

> Solution:
> * Empty alt property from `<img>`
> * Remove title property from `<svg>`

</details>

<details>
<summary>Report 3:

Main page should have a `<h1>`

</summary>

> Solution:
> * Add `<h1>` for page title
> * Convert large text of *I am Josua Fernando Simanjuntak* as `<h1>`

</details>

<details>
<summary>Report 4:

Heading in Blog Posts list (`/blog`) should not skip heading (no `<h2>`)

</summary>

> Solution:
> * Give a `<h2>` between `<h1>` and `<h3>` position
> * Assign the `<h2>` as section title for latest posts
> * Add another `<h2>` for recommended posts (optional, but will make structure better)

</details>

<details>
<summary>Report 5:

Each page should have only one `<main>` and do not place it as descendant of another `<main>`

</summary>

> Solution:
> * Remove one `<main>` that placed outside of main structure (placed in `app.js`)

</details>

<details>
<summary>Report 6:

A `<section>` must have a heading (`<h1>` - `<h6>`) to distinguish from the other sections

</summary>

> Solution:
> * Add a heading (`<h1>` - `<h6>`)
> * Use `<div>` if no heading suitable for the section

</details>

<details>
<summary>Report 7:

The id property should not be emptied

</summary>

> Solution:
> * Remove id property if no id suitable or needed

</details>

<details>
<summary>Report 8:

The trailing slash on the self-closed tags (void elements) does not have any function and can be removed

</summary>

> Solution:
> * Remove trailing slash from self-closed tag, such as `<meta>`, `<img>`, `<a>`, etc.

</details>
